File: Task-2/task2_ws/src/age_pkg/src/sub_age.cpp
#include <ros/ros.h>
#include "age_pkg/Age.h"

using namespace ros;

void callback(const age_pkg::Age::ConstPtr &age)
{
    ROS_INFO("\nMy age is %ld days, %ld months and %ld years", age->days, age->months, age->years);
}

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "sub_age", init_options::AnonymousName);
    NodeHandle nh;

    // subscribe
    Subscriber sub = nh.subscribe<age_pkg::Age>("age_topic", 1, callback);
    spin();

    return 0;
}

File: custom msg/custom msg_ws/src/age_pkg/scripts/sub_age.py
#!/usr/bin/python3

import rospy
from age_pkg.msg import Age


def callback(age):
    
    rospy.loginfo("\nMy age is %d days, %d months and %d years",age.days, age.months,age.years)



def sub_age():

    # node
    rospy.init_node("sub_node", anonymous=True)

    # subscribe
    sub = rospy.Subscriber("age_topic", Age, callback)
    rospy.spin()    


if __name__ == "__main__":
    sub_age()
